src/node/history.h

```diff
@@ -231,8 +231,15 @@ namespace ccf
 
     bool verify(HistoryTree* tree) const
     {
-      return tree->max_index() == path->max_index() && tree->root() == root &&
-        path->verify(root);
+      if (path->max_index() > tree->max_index())
+        return false;
+      else if (tree->max_index() == path->max_index())
+        return tree->root() == root && path->verify(root);
```

Comment:
```suggestion
      if (path->max_index() > tree->max_index())
      {
        return false;
      }
      else if (tree->max_index() == path->max_index())
      {
        return tree->root() == root && path->verify(root);
      }
```
This is what @achamayou was referring to - we try to avoid unscoped single-line blocks.